NastassiaBon/hometask-4/my_hash/my_hash_RSpec.rb

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+require_relative 'my_hash'
+require 'rspec'
+require 'pry'
+
+RSpec.describe MyHash do
+  subject(:my_hash) { described_class.new }
+
+  describe '#[]' do
+    before { my_hash[:key] = 2 }
+
+    context 'when key is missed' do
+      it "should return key's value" do
+        expect(my_hash[:key]).to eq(2)
+      end
+    end
+
+    context 'when the key went beyond the size of MyHash' do
+      it 'should return nil ' do
+        expect(my_hash[:a]).to eql(nil)
+      end
+    end
+
+    context 'when the key is skipped' do
+      it 'should show ArgumentError' do
+        expect { my_hash[] }.to raise_error ArgumentError
+      end
+    end
+  end
+
+  describe '#[]=' do
+    context 'when key and value are set' do
+      it 'should add value to MyHash' do
+        expect(my_hash.size).to eq(1)
```

Comment:
why 1? because you have added value in the previous test. But if I run this test separately - it will fall